src/utils/Processing.ts

```diff
@@ -564,6 +567,42 @@ export function writeFHIRResources(
   writeResources(outPackage.instances.filter(i => i._instanceMeta.usage !== 'Inline'));
 
   logger.info(`Exported ${count} FHIR resources as JSON.`);
+  return skippedResources;
+}
+
+export function writeFSHIndex(
+  outDir: string,
+  outPackage: Package,
+  inputDir: string,
+  skippedResources: string[] = []
+) {
+  const index: string[][] = [];
+  const otherIndex: any[] = [];
+  for (const [fileName, fshInfo] of outPackage.fshMap) {
+    if (!skippedResources.includes(fileName)) {
+      const relativeInput = path.relative(inputDir, fshInfo.file);
+      index.push([
+        fshInfo.fshName,
+        fshInfo.fshType,
+        relativeInput,
+        `${fshInfo.location.startLine} - ${fshInfo.location.endLine}`,
+        fileName
+      ]);
+      otherIndex.push({
+        fshFile: relativeInput,
+        fshName: fshInfo.fshName,
+        fshType: fshInfo.fshType,
+        startLine: fshInfo.location.startLine,
+        endLine: fshInfo.location.endLine,
+        outputFile: fileName
+      });
+    }
+  }
+  // write txt with nice formatting
+  index.unshift(['Name', 'Type', 'File', 'Lines', 'Output']);
+  fs.outputFileSync(path.join(outDir, 'fsh-generated', 'fsh-index.txt'), table(index));
+  // write json for machine usage
+  fs.outputJsonSync(path.join(outDir, 'fsh-generated', 'fsh-index.json'), otherIndex);
```

Comment:
I know this is for machines, but I think we should consider formatting the JSON so that humans can more easily parse it too. It takes a little more space, but I think it's worth it.